src/snippets.js

```diff
@@ -349,7 +349,7 @@ var SnippetManager = function() {
         return result;
     };
 
-    var processSnippetText = function(editor, snippetText, replaceRange) {
+    var processSnippetText = function(editor, snippetText, removeExtraIndent) {
```

Comment:
Do we expect users would want to change `removeExtraIndent` on the fly? I might be wrong but it feels like an option you'd want to set once for your project. If that's so, it might be cleaner to set the option once (e.g. at the `SnippetManager` level) and then take the option from there wherever we need it.

Reply:
Depending on implementation on consumer end, snippets might have different formatting and require indentation or not. E.g. if there are few different sources for snippets and one gives properly formatted snippet, while another not